### ISSUE TITLE
Type of tasks?

### ISSUE THREAD

**Issue by isaacmg** [amzn/metalearn-leap#1]
Hi thanks for the paper and providing code. I was wondering in your instructions you say `Require: criterion, model, tasks, opt_cls, meta_opt_cls, opt_kwargs, meta_opt_kwargs` What is the type of tasks? Originally I thought it was a nested list but lists don't have .sample() method. Also I assume criterion is `torch.nn.Modules.Loss` but what are the types of opt_cls and meta_opt_cls? Do you have a full example of wrapping an existing model in this fashion? 

Thanks.

**Comment by flennerhag** [comment]
Hi, thanks for reaching out! The algorithm is in pseudo-code, so ``tasks`` is only an abstraction. Typically ``criterion`` and ``model`` are ``torch.nn`` instances. ``opt_cls`` and ``meta_opt_cls`` are optimizer *classes* from ``torch.optim``, such as ``torch.optim.SGD``.  How to construct ``tasks`` depends on the problem. Typically, you have a list of dataloaders or environments and sample uniformly from them. A full example is given in the [Omniglot experiment](https://github.com/amzn/metalearn-leap/tree/master/src/omniglot), the ``tasks`` object is the ``DataContainer`` class defined [here](https://github.com/amzn/metalearn-leap/blob/master/src/omniglot/data.py).
